packages/framework/src/Command/RolesCommand.php

```diff
@@ -0,0 +1,393 @@
+use Shopsys\FrameworkBundle\Component\Security\Role\Section\AbstractRoleSectionProvider;
+use Shopsys\FrameworkBundle\Component\Security\Role\Section\RoleSection;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\Attribute\TaggedIterator;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Traversable;
+
+#[AsCommand(
+    name: 'shopsys:rbac:roles',
+    description: 'Display roles organized by context with their details and permissions',
+)]
+class RolesCommand extends Command
+{
+    private const DETAIL_VIEW_THRESHOLD = 5;
+
+    /**
+     * @var array<class-string<\Shopsys\FrameworkBundle\Component\Context\AbstractContext>, \Shopsys\FrameworkBundle\Component\Security\Role\Hierarchy\AbstractRoleHierarchyProvider>
+     */
+    private array $hierarchyProviders = [];
+
+    /**
+     * @var array<class-string<\Shopsys\FrameworkBundle\Component\Context\AbstractContext>,\Shopsys\FrameworkBundle\Component\Security\Role\Section\AbstractRoleSectionProvider>
+     */
+    private array $roleSectionsProvidersByContext;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Security\Role\RoleRegistryInterface $roleRegistry
+     * @param \Shopsys\FrameworkBundle\Component\Context\ContextResolverInterface $contextResolver
+     * @param \Symfony\Contracts\EventDispatcher\EventDispatcherInterface $eventDispatcher
+     * @param iterable<\Shopsys\FrameworkBundle\Component\Security\Role\Hierarchy\AbstractRoleHierarchyProvider> $hierarchyProviders
+     * @param iterable<class-string<\Shopsys\FrameworkBundle\Component\Context\AbstractContext>,\Shopsys\FrameworkBundle\Component\Security\Role\Section\AbstractRoleSectionProvider> $roleSectionsProviders
+     */
+    public function __construct(
+        private readonly RoleRegistryInterface $roleRegistry,
+        private readonly ContextResolverInterface $contextResolver,
+        private readonly EventDispatcherInterface $eventDispatcher,
+        #[TaggedIterator('shopsys.role_hierarchy_provider')]
+        iterable $hierarchyProviders,
+        #[TaggedIterator('shopsys.role_section_provider', defaultIndexMethod: 'getTargetContext')]
+        iterable $roleSectionsProviders = [],
+    ) {
+        parent::__construct();
+
+        foreach ($hierarchyProviders as $provider) {
+            $this->hierarchyProviders[$provider->getTargetContext()] = $provider;
+        }
+
+        $this->roleSectionsProvidersByContext = $roleSectionsProviders instanceof Traversable ? iterator_to_array($roleSectionsProviders) : $roleSectionsProviders;
```

Comment:
is this necessary? I would expect Symfony to inject the services always the same way, so it always (or never) be Traversable, or?

Reply:
Ok removed